dlc-node/src/main/scala/org/bitcoins/dlc/node/DLCDataHandler.scala

```diff
@@ -0,0 +1,74 @@
+package org.bitcoins.dlc.node
+
+import akka.actor._
+import akka.event.LoggingReceive
+import org.bitcoins.core.api.dlc.wallet.DLCWalletApi
+import org.bitcoins.core.protocol.tlv._
+
+import scala.concurrent._
+
+class DLCDataHandler(dlcWalletApi: DLCWalletApi, connectionHandler: ActorRef)
+    extends Actor
+    with ActorLogging {
+  implicit val ec: ExecutionContextExecutor = context.system.dispatcher
+
+  override def preStart(): Unit = {
+    val _ = context.watch(connectionHandler)
+  }
+
+  override def receive: Receive = LoggingReceive {
+    case lnMessage: LnMessage[TLV] =>
+      log.info(s"Received LnMessage ${lnMessage.typeName}")
+      lnMessage.tlv match {
+        case msg @ (_: UnknownTLV | _: DLCOracleTLV | _: DLCSetupPieceTLV) =>
+          log.error(s"Received unhandled message $msg")
+        case _: InitTLV =>
+          () // todo init logic
+        case error: ErrorTLV =>
+          log.error(error.toString)
+        case ping: PingTLV =>
+          val pong = PongTLV.forIgnored(ping.ignored)
+          connectionHandler ! LnMessage(pong)
+        case pong: PongTLV =>
+          log.debug(s"Received pong message $pong")
+        case dlcOffer: DLCOfferTLV =>
+          val _ = for {
```

Comment:
so it isn't ideal that we are dropping all these `Future` on the floor. We have no way of logging an error that occurs from these. 

i've written a patch to improve this and add a log

https://github.com/Christewart/bitcoin-s-core/commit/281e6c2f1e2ac5c53db92552805f4eba1a074323

Reply:
I've also got a patch for thread pool leaks: https://github.com/Christewart/bitcoin-s-core/tree/2021-07-30-dlc-threadpool-leaks

Send me a DM on slack if you want me to open these as PRs on your repo, otherwise they should be easy enough to cherry pick